homeassistant/components/ruckus_unleashed/config_flow.py

```diff
@@ -19,22 +19,24 @@
 DATA_SCHEMA = vol.Schema({"host": str, "username": str, "password": str})
 
 
-def validate_input(hass: core.HomeAssistant, data):
+async def validate_input(hass: core.HomeAssistant, data):
     """Validate the user input allows us to connect.
 
     Data has the keys from DATA_SCHEMA with values provided by the user.
     """
 
     try:
-        ruckus = Ruckus(data[CONF_HOST], data[CONF_USERNAME], data[CONF_PASSWORD])
+        ruckus = await Ruckus.create(
+            data[CONF_HOST], data[CONF_USERNAME], data[CONF_PASSWORD]
+        )
     except AuthenticationError as error:
         raise InvalidAuth from error
     except ConnectionError as error:
         raise CannotConnect from error
 
-    mesh_name = ruckus.mesh_name()
+    mesh_name = await ruckus.mesh_name()
 
-    system_info = ruckus.system_info()
+    system_info = await ruckus.system_info()
     try:
         host_serial = system_info[API_SYSTEM_OVERVIEW][API_SERIAL]
     except KeyError as error:
```

Comment:
Although you didn't chance it from before I'd expect the await to be inside the try as well (and the mesh_name as well, assuming both are I/O functions given you are awaiting them).